src/QtImageReader.cpp

```diff
@@ -84,21 +84,16 @@ void QtImageReader::Open()
 			ZmqLogger::Instance()->AppendDebugMethod("QtImageReader::Open", "ba.size()", ba.size(), "", -1, "", -1, "", -1, "", -1, "", -1);
 
 			// use byte array as data source for resvg
-			char *error = nullptr;
-			rtree = resvg_parse_rtree_from_data(ba.constData(), &opt, &error);
+			int error = 0;
+			error = resvg_parse_tree_from_data(ba.constData(), ba.size(), &opt, &rtree);
 			if (!rtree) {
-				ZmqLogger::Instance()->AppendDebugMethod("QtImageReader::Open", "resvg_parse_rtree_from_data error", -1, error, -1, "", -1, "", -1, "", -1, "", -1);
-				resvg_error_msg_destroy(error);
+				ZmqLogger::Instance()->AppendDebugMethod("QtImageReader::Open", "resvg_parse_tree_from_data error", error, "", -1, "", -1, "", -1, "", -1, "", -1);
```

Comment:
You should probably change this to check `if (error)` instead of `if (!rtree)`, since it looks like the semantics of the `resvg_parse_[r]tree_from_data()` function(s) are different. It's conceivable that `rtree` might be non-null even when there is an error.

Also, if `error` is an integer, logging it as-is won't be very meaningful. Is there a utility function to translate the return values from `resvg_parse_tree_from_data` into meaningful error messages? (Or are they just ERRNOs or something standard like that? Then you can use `strerror(error)` to convert to a useful error message string.)

Reply:
Ah, I see how the semantics of `AppendDebugMethod()` work now, and why `error` would be logged "raw". (Though, if it is possible to turn the error value into a string with `strerror()` or similar, I would think that would still be useful as an additional component of the log message.)